compiler/rustc_typeck/src/coherence/orphan.rs

```diff
@@ -330,6 +344,24 @@ fn emit_orphan_check_error<'tcx>(
     })
 }
 
+fn emit_orphan_check_for_impl_on_types<'tcx, T: rustc_errors::EmissionGuarantee>(
+    full_impl_span: Span,
+    self_ty_span: Span,
+    ptr_ty: &ty::TypeAndMut<'_>,
+    diag: &mut DiagnosticBuilder<'tcx, T>,
+) {
+    let mut_key = if ptr_ty.mutbl == rustc_middle::mir::Mutability::Mut { "mut " } else { "" };
+    let msg_sugg = "consider to use a new wrapper type".to_owned();
+    let sugg = vec![
+        (
+            full_impl_span.shrink_to_lo(),
+            format!("struct WrapperType(*{}{});\n\n", mut_key, ptr_ty.ty),
+        ),
+        (self_ty_span, "WrapperType".to_owned()),
+    ];
+    diag.multipart_suggestion(msg_sugg, sugg, rustc_errors::Applicability::MachineApplicable);
```

Comment:
This is not `MachineApplicable` -- I think `MaybeIncorrect` is more accurate.

Reply:
Mh! yes I think this is the safes option!